### PR TITLE
 JAMES-2866 MockSMTP server behaviors setup over HTTP (Jetty)

### PR DESCRIPTION
**Pull request by chibenwa**
Same than https://github.com/linagora/james-project/pull/2615 , but using Jetty HTTP server

### CODE REVIEW

**Comment by trantienduchn** [comment]
test this please, github connecting issue

**Comment by trantienduchn** [comment]
```
[0bc2ae936b993d4ca6bdaf0e667a7c87a8f1090f] [ERROR] Errors: 
[0bc2ae936b993d4ca6bdaf0e667a7c87a8f1090f] [ERROR]   MemoryUploadCucumberTest » Socket Broken pipe (Write failed)
```

**Comment by trantienduchn** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[0bc2ae936b993d4ca6bdaf0e667a7c87a8f1090f] [ERROR] Errors: 
[0bc2ae936b993d4ca6bdaf0e667a7c87a8f1090f] [ERROR]   AccessTokenRepositoryTest.validTokenMustBeRetrieved:45 » InvalidAccessToken df...
```
test this please

**Comment by trantienduchn** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
MemoryUploadCucumberTest » Socket Broken pipe (Write failed)

test this please

**Comment by chibenwa** [comment]
test this please (restart)

File: server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        .then()
+            .body(".", hasSize(0));
+    }
+
+    @Test
+    void putShouldReturnNoContent() {
+        given()
+            .body(JSON_BEHAVIORS)
+        .when()
+            .put()
```
Comment (remk) [inline server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java]: In the gitlab tickets you used the HTTP verb `POST` which seemed to be the right choice given that the resource is not identified in the URI. Why did you choose to use the `PUT` verb ?

Comment (remk) [reply server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java]: Ok it's for updating the list of behavior so i guess `PUT` is the right choice. Could you just update the gitlab ticket to be coherent with your new decision ?

Comment (chibenwa) [reply server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java]: > Ok it's for updating the list of behavior and it's idempotent so PUT is the right choice. 

That's it!

> Could you just update the gitlab ticket to be coherent with your new decision ?

Sure <3

**Comment by chibenwa** [comment]
I rebased on the POJO, my work included 2 fixes:

  - Operator matchAll was being passed a dummy value. I reworked class structure (and move matchAll concept at the condition level) to not require this hack.
 - Behaviors' condition do not need to be optional now that we have a "matchAll" condition

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**